src/app/services/error/error-message.service.ts

```diff
@@ -0,0 +1,52 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { ErrorMessage } from '@darts-types/error-message.interface';
+import { HeaderService } from '@services/header/header.service';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+//Contains endpoints where errors are handled in their component
+const subscribedEndpoints = ['/api/cases/search'];
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ErrorMessageService {
+  private errorMessage: BehaviorSubject<ErrorMessage | null> = new BehaviorSubject<ErrorMessage | null>(null);
+  readonly errorMessage$: Observable<ErrorMessage | null> = this.errorMessage.asObservable();
+
+  constructor(
+    private headerService: HeaderService,
+    private router: Router
+  ) {}
+
+  handleErrorMessage(error: HttpErrorResponse) {
+    error.error && this.setErrorMessage({ status: error.status, type: error.error.type });
```

Comment:
What's `error.error` && for?

Reply:
The current consuming component search depends on error.error.type and this is just a check to ensure error object exists before checking as it can be null